### ISSUE TITLE
add att-dependency class to store UD (and similar) information beyond the word level

### ISSUE THREAD

**Issue by bansp** [LingSIG/TEI#6]
This is a task formulated at the meeting held on 23-oct-2024.

Piotr will create a stub of a class that builds upon att-linguistic to add `@head` and `@deprel` (names not set in stone yet), for encoding dependency information in the same manner as the general morphosyntactic info, and thus to have a handy format for the conversion of dependency treebanks. Probably also `@index` may be needed, unless we decide to use `@n` (but it might be sensible to leave the `@n` for project-internal purposes).
Afterwards, Piotr and Andreas are going to build the documentation for the class (everyone's input is very welcome!), so that it can be submitted to the Council for consideration.

**Comment by bansp** [comment]
## Links
The test versions are now at

- (Spec) https://jenkins-paderborn.tei-c.org/view/LingSIG/job/TEIP5-LingSIG-att.linguistic.dependency/lastSuccessfulBuild/artifact/P5/Guidelines-web/en/html/ref-att.linguistic.dependency.html
- (Chapter) https://jenkins-paderborn.tei-c.org/view/LingSIG/job/TEIP5-LingSIG-att.linguistic.dependency/lastSuccessfulBuild/artifact/P5/Guidelines-web/en/html/AI.html#AILADEP

## Issues
There are several issues worth discussing or at least noting:

1. It does not seem possible to reference graphics by doing something like `../Guidelines(...)/Images/(...)` -- such paths are filtered out as invalid, and I didn't want to try setting up a new directory for images under /Source just for the LingSIG -- I guess that the idea might not be accepted so easily. (I wanted to use a graph diagram to explain what the particular new attributes are supposed to do, but it seems that that can only be explained in the body of the Guidelines, where images are legal.)
2. We should be able to show the source CoNNL in the documentation, and I've tested displaying it by means of 
    *  a `<table>` element (totally clunky, unreadable and needs some magic to render columns nicely);
    * a `p` with appropriate monospace styling and xml:space setting, so that the plain text displays sensibly (but it doesn't);
    * an XML comment inside `egXML` (and ran into some rendering issues; the entire `exemplum` would fail to render) but, semantically, this smells of abuse, so I've also tried using `eg`, and **that is our winner**!

3. I think there is no escaping the addition of `xpos` and `deps`. The first because of the power that it offers (why would we impoverish the data so much) and the second, well, basically for the same reason: many extended graphs are floating around, and not being able to transform them into TEI would mean cutting off part of the potentially interested audience
4. One could think of adding Schematron constraints on some parts of the representation, but at that point, we would really be doing a UD-oriented encoding, whereas it might be worth it to make it a bit more open, and to push UD-well-formedness checks onto the encoder/user.

## Serialisation
Serialisation of the UD/CoNNL-U format:
| UD column | TEI representation | remarks |
| --- | --- | --- |
| ID | `@n` | att.global |
| FORM | element content | |
| LEMMA | `@lemma` | att.linguistic |
| UPOS | `@pos` | att.linguistic |
| XPOS | `@xpos` | postulated |
| FEATS | `@msd` | att.linguistic |
| HEAD | `@head` | att.linguistic.dependency |
| DEPREL | `@deprel` |  att.linguistic.dependency |
| DEPS | `@deps` | postulated |
| MISC | ? | |
| SpaceAfter=No in MISC | `@join="right"` | att.linguistic |

**Comment by bansp** [comment]
I'd say let us push on and add `@xpos` and `@deps`, and maybe think of the data types for those attributes.

(Why not add `@upos`, someone may ask: because UPOS is the obligatory part, so it seems that the existing `@pos` is the right serialisation for it, as opposed to the optional `@xpos`, which is an extra that comes as a bonus with UD. So it makes sense to define it where other UD-defined attributes are.)

**Comment by nolda** [comment]
I would like to propose to change the description of `@head` as follows (presupposing the we are using `@n` for marking up the index of the head):

> encodes a dependency relation between the element bearing this attribute and the element having the attribute value as its index (i.e., as the value of its attribute `<att>n</att>`). In a dependency graph, the node represented by the former element is said to be the `<term>dependent</term>` and the node represented by the latter is said to be its `<term>head</term>`. By convention, the value `<val>0</val>` indicates that the element bearing this attribute is the root of the dependency graph.

**Comment by nolda** [comment]
And likewise for the description of `@deprel`:

> provides a classifying label for the dependency relation between the element bearing this attribute and the element identified by the attribute `<att>head</att>`. If the value of `<att>head</att>` is `<val>0</val>`, then, by convention, the value of `<att>deprel</att>` should be `<val>root</val>`.

**Comment by bansp** [comment]
These are very elegant and to the point -- thank you, Andreas! I'll push them asap, so that we can all admire them tomorrow :-)

**Comment by bansp** [comment]
(This is just a note: our build is broken because the current TEI build is broken. It took quite a bit of trying before it occurred to me to check. Weird, I would normally expect that the main branch would be prevented from entering such a state by checking new features in separate branches before they are merged and potentially kill a lot of dependent projects. But... it is as it is.)

**Comment by bansp** [comment]
Since the TEI build is still broken, and I would prefer to keep the att.linguistic.dependency branch completely in sync with the main TEI tree, below is a temporary alternative, from which I have removed the offending commits:

* Guidelines: https://jenkins-paderborn.tei-c.org/view/LingSIG/job/TEIP5-LingSIG-tests/lastSuccessfulBuild/artifact/P5/Guidelines-web/en/html/AI.html#AILADEP
* Spec: https://jenkins-paderborn.tei-c.org/view/LingSIG/job/TEIP5-LingSIG-tests/lastSuccessfulBuild/artifact/P5/Guidelines-web/en/html/ref-att.linguistic.dependency.html

**Comment by bansp** [comment]
(we're back to the original branch)

**Comment by bansp** [comment]
Here are our new links:

- Guidelines: https://jenkins-paderborn.tei-c.org/view/LingSIG/job/TEIP5-LingSIG-att.linguistic.dependency/lastSuccessfulBuild/artifact/P5/release/doc/tei-p5-doc/en/html/AI.html#AILADEP
- Spec: https://jenkins-paderborn.tei-c.org/view/LingSIG/job/TEIP5-LingSIG-att.linguistic.dependency/lastSuccessfulBuild/artifact/P5/release/doc/tei-p5-doc/en/html/ref-att.linguistic.dependency.html

The reason for the modification is that now, we can use a modified schema (p5.nvdl) that knows about the added attribute class, so that everything in oXygen is green rather than red as before. (I know that that explanation is a bit condensed... :-) )

**Comment by nolda** [comment]
> Why not add `@upos`, someone may ask: because UPOS is the obligatory part, so it seems that the existing `@pos` is the right serialisation for it, as opposed to the optional `@xpos`, which is an extra that comes as a bonus with UD. So it makes sense to define it where other UD-defined attributes are.

I fully agree that UPOS should be encoded as `@pos`. As to XPOS, I would not try to cope for it by means of the att-dependency class: XPOS does not represent linguistic dependency relations between tokens but rather provides an alternate POS classification of tokens. Using concurrent POS tagging or not is orthogonal to the markup of dependencies. So, I would propose to restrict the att-dependency class to attributes for linguistic dependencies proper and to postpone the question of representing concurrent POS tagging.

**Comment by nolda** [comment]
One could imagine various solutions for representing both UPOS *and* XPOS. Two straightforward solutions would be:
- providing a list value for `@pos`,
- defining an additional attribute, say `@pos.alt`, for the XPOS value.

Whatever the preferable solution might be – the representation of concurrent POS tagging is, in principle, independent from the representation of syntactic dependencies.

**Comment by nolda** [comment]
As to the markup of the root, one might envisage three alternatives:

1. with an abstract, unencoded root node, pointed to from an element with `@head=0` (strictly parallel to UD):
```
<w n="1" head="2" deprel="nsubj" ...>I</w>
<w n="2" head="0" deprel="root" ...>had</w>
<w n="3" head="4" deprel="det" ...>no</w>
<w n="4" head="2" deprel="obj" ...>clue</w>
<pc n="5" head="2" deprel="punct" ...>.</pc>
```
2. marking up the parent span with `@n=0`, e.g.:
```
<s n="0">
<w n="1" head="2" deprel="nsubj" ...>I</w>
<w n="2" head="0" deprel="root" ...>had</w>
<w n="3" head="4" deprel="det" ...>no</w>
<w n="4" head="2" deprel="obj" ...>clue</w>
<pc n="5" head="2" deprel="punct" ...>.</pc>
</s>
```
3. leaving the root implicit (as in traditional dependency grammar):
```
<w n="1" head="2" deprel="nsubj" ...>I</w>
<w n="2" ...>had</w>
<w n="3" head="4" deprel="det" ...>no</w>
<w n="4" head="2" deprel="obj" ...>clue</w>
<pc n="5" head="2" deprel="punct" ...>.</pc>
```

In the 3rd, minimalistic, alternative, the root is implicitly determined by the dependency tree graph and can heuristicly be determined from the fact that there is no `@head` on the verb node.
All of the alternatives would be, given the availability of the att-dependency class, valid TEI. Still, it would be useful to recommend one of them.

**Comment by bansp** [comment]
As for `@pos`, I agree that, from the dependency perspective, the xpos is a different kettle of fish. On the other hand, if we want the serialisation to be taken up relatively smoothly, I think we should keep the information around, and rather close by.

One option (that I'm not super happy with, but it's some kind of compromise) is to push the extra layer into the grammar of the value of `@pos`, namely to serialise upos and xpos inside the same attribute, separated by... a separator. Let us assume that the separator is a colon. Then, an example would be `pos="NOUN:NN"`, and the serialisation principle would be something like:

## Serialisation
Serialisation of the UD/CoNNL-U format:
| UD column | TEI representation | remarks |
| --- | --- | --- |
| UPOS | `substring-before(@pos,':')` | att.linguistic |
| XPOS | `substring-after(@pos,':')` | att.linguistic |

Would that work as a proposal for discussion by the Council?

**Comment by bansp** [comment]
Just a quick pre-meeting note about the head reference: I would vote for leaving it as close to UD as possible, i.e. like in example one above: `@deprel` preserved where it is used in CoNLL, and the matching of `@head` and `@n` left outside the encoding, for the interpreting software.

**Comment by bansp** [comment]
And a quick post-meeting note about the separator: it has been suggested that the pipe symbol is used as a separator (rather than the colon), as our best practice suggestion.
I was a bit hesitant because the pipe is used in the feats (our `@msd`) to aggregate (or 'add') the individual feature-value statements into some kind of matrix, whereas we would be suggesting, when using it inside `@pos`, that it signals _disjunction_ of the values on both sides.
But, perhaps, we can also say that the disjunction is not very strong, and that `NOUN|NN` can also be understood as aggregation.

**Comment by bansp** [comment]
I'm going to sketch the issue for the TEI-C here at first, to make sure that I have not omitted anything relevant.

I will assume that the modified chapter and the additional spec speak for themselves or at least illustrate the theoretical side of the issue relatively comprehensively, so I will focus on the technicalities that the Council may possibly wonder about.

Something we should bear in mind and warn the Council about is that, in order to validate our additions, we had to target the document grammar that knew about the extra spec file, hence the following at the top of our files:
```xml
<?xml-model href="https://jenkins-paderborn.tei-c.org/view/LingSIG/job/TEIP5-LingSIG-att.linguistic.dependency/lastSuccessfulBuild/artifact/P5/release/xml/tei/odd/p5.nvdl" type="application/xml" schematypens="http://purl.oclc.org/dsdl/nvdl/ns/structure/1.0"?>
```

Maybe I should add the regular vanilla PI in a comment, so that the council has everything on the plate.

Also, intriguingly, the current spec errors out on my disk, due to a Schematron assertion failure: doubling `<remarks xml:lang="en"`. I'm a bit puzzled by that, because I would think that the error should prevent the branch from building, but I may be missing something.

So, I'm going to fix the `remarks` and add the vanilla schema-association PIs, commented out. And then I'll proceed with the suggested text of the TEI-C issue accompanying the PR.

The changed or new files that we are going to submit in the PR are:

- att.linguistic.dependency.xml                           (new)
- att.linguistic.xml                                                  (modified to hook the above into the class system)
- Guidelines/en/AI-AnalyticMechanisms.xml    (section and hooks added)
- images under /Guidelines/en/Images/           (new)

**Comment by bansp** [comment]
I will modify the definitions of attributes. Both need to mention the conflation of basic and extended dependencies. I am not sure if we can do without additional prose about that.

Also, there is one backwards-compatibility issue that has now occurred to me in connection with `@pos`: with the potential storage of UPOS and XPOS in a single attribute, as whitespace-separated values, we may be breaking some existing tools or workflows, which process `@pos` as a single atomic value. That could be avoided if we use the other approach that we have considered, namely glueing the two values with a pipe, thus ending up with "CCONJ|CC" instead of "CCONJ CC", and "NOUN|NNS" instead of "NOUN NNS".

To any legacy tools or workflows, pipe-separated values would just be a complex POS value, processed without error. But many XSLT-based scripts may error out when encountering a sequence instead of an atomic value (thus far, they haven't had any need to check for sequences).

How about the following: 

- no change in the datatype of POS
- | as the separator
- clear statement of both options for the Council, so that they decide which they think is better.

The potential counter is that the pipe-separated value of `@msd` parses into a bag of features, whereas the pipe-separated value of `@pos` should parse into an ordered set. But they are separate attributes, and it's not at all uncommon for different complex attributes to have different associated parsing rules.

Another thing I'm wondering about is a Schematron check, making sure at least that the number of values in `@head` equals the number of values in `@deprel`.

I realise that I am digging up stuff that we've already buried, but we may well be asked to motivate the datatype change for `@pos` and the various other conventions that we propose, and I think it's better to have all of that out in the open, for the Council to be able to offer their insights and suggestions.

**Comment by nolda** [comment]
> Also, there is one backwards-compatibility issue that has now occurred to me in connection with `@pos`: with the potential storage of UPOS and XPOS in a single attribute, as whitespace-separated values, we may be breaking some existing tools or workflows, which process `@pos` as a single atomic value. That could be avoided if we use the other approach that we have considered, namely glueing the two values with a pipe, thus ending up with "CCONJ|CC" instead of "CCONJ CC", and "NOUN|NNS" instead of "NOUN NNS".
> To any legacy tools or workflows, pipe-separated values would just be a complex POS value, processed without error. But many XSLT-based scripts may error out when encountering a sequence instead of an atomic value (thus far, they haven't had any need to check for sequences).
> How about the following:
> *   no change in the datatype of POS
> * | as the separator
> *  clear statement of both options for the Council, so that they decide which they think is better.

I agree that changing the datatype of  `@pos` from `teidata.text` to list of words (sequences of `teidata.word`) may raise backward-compatibilty issues. Perhaps, using a space here as a separator between concurrent UPOS and XPOS tags is not optimal anyway: in `@head` and `@deprel`, we are using list values not for concurrent, but for additional dependency relations.

In my view, `|` is not a good choice either, given the different semantics the pipe separator has in CoNLL-U FEATS, which we propose to use verbatim in `@msd`. So how about a dedicated separator for concurrent markup, say `:`?

**Comment by bansp** [comment]
A colon is a notorious offender in XML world, but we don't need to bother about that in this context, and I have also thought it very fitting as a separator. But before moving on, let's have one more thought experiment. (It's not a waste of time, I think, because we're recording the results in one place and we're rehashing our older arguments in the process.)

What if we assigned a concrete parsing function to the separator. and said that: 

**we treat `|` as a separator for unordered values, and `:` as a separator for ordered values**   

 The unordered case is clear. But then, we'd have:

#### 1. "NOUN:NSS"

| | item 1 | item 2 |
| --- | --- | --- |
`@pos`| NOUN | NNS |
parse:| UPOS="NOUN" | XPOS="NNS" |

#### 2. `@head` and `@deprel`
```xml
<w n="1" head="2:4" deprel="nsubj:nsubj" pos="PRON:PRP" msd="Case=Nom|Number=Plur" lemma="they">They</w>
```
| | item 1 | item 2 |
| --- | --- | --- |
`@head`| 2 | 4 |
`@deprel`| nsubj | nsubj |
parse:| arc to 2 is labelled 'nsubj' | arc to 4 is labelled 'nsubj'|

<br /> <br /> 

```xml
<w n="4" head="2:0" deprel="conj:root"  pos="VERB:VBP" msd="Number=Plur|Person=3|Tense=Pres" lemma="sell">sell</w>
```
| | item 1 | item 2 |
| --- | --- | --- |
`@head`| 2 | 0 |
`@deprel`| conj | root |
parse:| arc to 2 is labelled 'conj' | arc to 0 is labelled 'root'|

<br /> 

You say, above:
> in `@head` and `@deprel`, we are using list values not for concurrent, but for additional dependency relations

... but I am not convinced that -- somewhat paradoxically, perhaps -- the terms couldn't be reversed without causing interpretive issues:  XPOS is an _additional_ value, while "2" and "0" as well as "conj" and "root" are pairwise _concurrent_.

In other words, aren't we really looking at analogous constructions? And if so, perhaps we could suggest the rule above (while potentially also sketching other options)?

(BTW. Earlier in the chapter, we see:
```xml
<w pos="PPER" lemma="wir" msd="pers:subst:p1:nom:pl">Wir</w>
```

-- and that is fine, because the colon separates ordered values, here.)

**Comment by nolda** [comment]
> Earlier in the chapter, we see:
> ```xml
> <w pos="PPER" lemma="wir" msd="pers:subst:p1:nom:pl">Wir</w>
> ```
> -- and that is fine, because the colon separates ordered values, here.

I would object to this. Fn. 83 in [chap. 18 of the Guidelines](https://tei-c.org/release/doc/tei-p5-doc/en/html/AI.html#AILALW) says about this example: "The annotation values have been adapted from the [CLARIN Weblicht service](https://weblicht.sfs.uni-tuebingen.de/weblicht/), where e.g. the full morphosyntactic description of the first item reads: `[cat pronoun, personal true, substituting true, person 1, case nominative, number plural]`, and has been mapped from a sequence of attribute-value pairs suitable for feature structure notation, into a compressed form that fits inside a single attribute value." In mathematical terms, a feature structure essentially is an (unordered) set of attribute-value pairs. If the `@msd` value here is indeed to be interpreted as being equivalent to (the values in) a feature structure, then this value is unordered, too.

In the [`att.linguistic` spec](https://tei-c.org/release/doc/tei-p5-doc/en/html/ref-att.linguistic.html), dots are used for more or less the same purpose:
```xml
<ab>
 <w pos="PPER" msd="1.Pl.*.Nom">Wir</w>
 <w pos="VVFIN" msd="1.Pl.Pres.Ind">fahren</w>
 <w pos="APPR" msd="--">in</w>
 <w pos="ART" msd="Def.Masc.Akk.Sg">den</w>
 <w pos="NN" msd="Masc.Akk.Sg">Urlaub</w>
 <pc pos="$." msd="--">.</pc>
</ab>
```

So in the existing documentation of `att.linguistic`, there appears to be no recommended separators nor a consistent use thereof.

**Comment by bansp** [comment]
Meeting resolutions:

- we are going to sketch the TEI issue here, as one of the comments
- Piotr is going to post it at TEI-C and to set up the accompanying PR from a branch of our TEI-dev mirror
- The issue is going to have minimal lead text, so that the audience is not overwhelmed, because we are also going to reference the proposed text of the chapter and the spec
- we will mention the addition of extra attributes and the role of the separator: whitespace, which is expected to indicate that the separated values are ordered

**Comment by nolda** [comment]
Here is a first proposal for the TEI issue’s initial comment, possibly restricting it to the content of the "Overview" section, and using the text in the "Discussion" section for a follow-up comment.

---

### Overview

This PR proposes to add a new attribute class called `att.linguistic.dependency` with attributes `@head` and `@deprel`, providing a lightweight means for annotating syntactic dependency relations between corpus tokens in the content of word-level elements `<w>` and `<pc>`. This new attribute class extends the existing attribute class `att.linguistic` for lightweight linguistic annotation, complementing the latter's attributes `@lemma`, `@lemmaRef`, `@pos`, `@msd`, and `@join`. It thereby allows for encoding not only the output of lemmatisers, part-of-speech taggers, and morphological analysers, but also that of dependency parsers in a straightforward, simple way.

### Discussion

As illustrated in the proposed Guidelines section 18.4.3 on "Syntactic dependency relations between word-level elements", syntactic dependency relations such as 'subject' or 'object' between words and other tokens can be modelled by annotating each token with a reference to its syntactic head (if any) and with a label that states the linguistic type of the dependency. Such an approach also underlies the widely used column-based CoNLL-U format, where each token is annotated, *inter alia*, with a numerical index in the `ID` column, a reference to the index of its head in the `HEAD` column, and a label of the type of the dependency relation in the `DEPREL` column. In TEI XML with the proposed addition of the `att.linguistic.dependency` attribute class, dependency relations can be encoded by using the `@n` attribute of `<w>` and `<pc>` elements for the numerical indices of the tokens in their content, while the `@head` attribute references the numerical indices of their heads and the `@deprel` attribute gives the corresponding type of the dependency relation.

This annotation scheme is not restricted to bare dependency structures, but is also capable of representing the content of other CoNLL-U columns, including `LEMMA` for lemmas, `UPOS` and `XPOS` for part-of-speech tags, `FEATS` for morphosyntactic features, and `DEPS` for enhanced syntactic dependencies, the latter being encoded as list items, separated by whitespace. (For an overview, cf. the table in the specification of the proposed attribute class and the examples discussed in section 18.4.3.) Given the lightweight nature of the annotation scheme, XML IDs are avoided here, which would have to be unique across an XML document, while CoNLL-U indices only need to be unique within the scope of a sentence; rather CoNLL-U IDs and its other columns can be directly mapped to attribute values in TEI XML.

**Comment by nolda** [comment]
A brief addendum to the delimiter discussion above.
There happens to be at least one POS tagset which uses `:` as a POS tag, viz. the [Penn Treebank tagset](https://www.sketchengine.eu/penn-treebank-tagset/), where `:` is used as a tag for a class of certain punctuation characters. So using `:` as a delimiter in `@pos` between UPOS and XPOS tags would be problematic here.
While I am not aware of any tagsets using `|` as a POS tag, it is not excluded in principle.
So I propose to stick to whitespace as a separator between list items in attribute values, as originally proposed.

**Comment by bansp** [comment]
Thanks, Andreas. 
As we agreed, I will attempt not to add much beyond the lead and the basic technical solutions that we've adopted, so that the information overhead / redundancy is minimal. I will mention UD/CoNLL-U but I don't think we need to agree on exactly how I'm going to do that, word by word (that would be overkill, and whatever I end up posting can be edited and commented on just as well).

I'm going to check the technicalities now, including Mr. Jenkins and whether our branch needs to get updated, and I hope to be ready within two hours or less.

**Comment by bansp** [comment]
There is one issue that, perhaps, is nevertheless worth handling before the submission, just to avoid an unnecessary fork in the discussion: we explicitly base our solution on UD (while stressing that the aim is more general). The UD flavour of dependency graphs apparently prefers to point at the dependants, rather than the head. Which is, on at least one level, counterintuitive with respect to the names of attributes (though I see some space for discussion). I've looked at the UD documentation, the currently preferred print reference, and the output of some tools, and they all agree to represent dep diagrams using arrows exactly reversed with respect to our arrows.

So, we point _at_ the dependency  head in the XML representation, but in the visual diagram, the arrows point in the opposite direction, _from_ the head. 

@nolda Would you consider reversing the arrows in the affected diagrams? We could then finalise the PR next week.

**Comment by bansp** [comment]
A very quick note after the presentation:
* Our use of `@n` did not sit well with some in the audience (notably @sydb), because it is imaginable (especially in the scenarios of enrichment) it is already taken by some other information in the given resource.
* Our name of the `@head` attribute raised doubt due to it being relatively generic.
* The arguments I have given against using `@xml:id` were explicitly accepted by Syd (and, implicitly, by others).

I did not have the time to mention our suggestion to pack `DEPS` into a sequential package serialised inside `@head` and `@deprel`, or to split `@pos` into the `UPOS`/`XPOS` stream. I occasionally think that a shift from explicit listing inside the putative `@deps` to implicit matching of positions inside the two sequential attributes may go against the assumption that we want to provide a smooth way for NLP colleagues to adopt our TEI-based format. It is of course a trade off but it might be that one more attribute that simply copies the content of `DEPS` and can be processed with the same algorithm as `DEPS` is, miiiight be the simpler solution, overall. And whether we add two or three (or four -- see below) attributes is not so very important, given that we add them in a very well defined bundle.

Summing up, here's a more 'explosive' suggestion, based on the feedback to the talk and on the above considerations:
* `@dephead` (butt-ugly, I admit -- but an actual suggestion from the room); hmm... `@depHead`?
* `@depindex` (`@depIndex`?) -- for `@n`
* `@deps` -- as above
* `@deprel` potentially becoming `@depRel` to keep to the camel nature of TEI naming
* `@pos` -- let's assume it stays as a potentially two-membered value

Question (from me, not from the audience): don't we care about full reproducibility of the input (say) UD markup? If we do then we need a box for the content of the `MISC` column (currently, we only single out (and transform) the value of `@join` from there. That would necessitate something like `@depMisc`...

**Comment by bansp** [comment]
Andreas and I have worked on this issue since September and we feel that it is ripe for sharing with the LingSIG and for a submission to the Council. The links to our proposal are unchanged; the are listed [at the top of this note](https://github.com/LingSIG/TEI/issues/6#issuecomment-2483258283).

**Comment by bansp** [comment]
Riding the wave of putting together minutes of the last LingSIG meeting, I decided to finally prepare and post the overdue PR, and here are the links:

PR: https://github.com/TEIC/TEI/pull/2888

issue: https://github.com/TEIC/TEI/issues/2889
